File: matrix.exponential/R/matrix.exponential.v8.R

source("matrix.exponential.R")

library(parallel)

n <- 2000
m <- 100
s <- 0

set.seed(0)

X1 <- matrix(runif(n^2),n,n)
X2 <- matrix(runif(n^2),n,n)

# detectCores()

result <- mclapply(list(X1,X2),matexp,m,mc.cores=4)

File: matrix.exponential/python/matrix.exponential.v8.py
from matexp import matexp
import numpy as np
from multiprocessing import Pool

n = 2000
m = 100
s = 0

np.random.seed(s)

pool = Pool(2)

X1 = np.random.rand(n,n)
X2 = np.random.rand(n,n)

print("starting Y1 and Y2")
result = pool.starmap(matexp,[(X1,m),(X2,m)])
















